### PR TITLE
Let speakers settle in L2 ExternalTrafficPolicy=Local tests 

### PR DESCRIPTION
**Pull request by oribon**
Since we're creating multiple pods, the announcing
speaker can change until all the pods are ready and
the situation is stabilized.
Adding a 5 seconds sleep should be enough for the
speakers to settle on who announces the service,
and should be enough for it to complete sending the
gratuitous ARPs as that's the time set in spamLoop().

This should solve the flake we're experiencing recently in the CI.

Signed-off-by: Ori Braunshtein <obraunsh@redhat.com>

### CODE REVIEW

File: e2etest/layer2_test.go [hunk 1]
```diff
@@ -120,6 +120,9 @@ var _ = ginkgo.Describe("L2", func() {
 			epNodes, err := jig.ListNodesWithEndpoint() // Only nodes with an endpoint could be advertising the IP
 			framework.ExpectNoError(err)
 
+			// Give the speakers enough time to settle and for the announcer to complete its gratuitous.
+			time.Sleep(5 * time.Second)
```
Comment (fedepaol) [inline e2etest/layer2_test.go:124]: Can we leverage eventually / consistently instead of the sleep?

Comment (oribon) [reply e2etest/layer2_test.go:124]: Not sure, what would we put there?

Comment (fedepaol) [reply e2etest/layer2_test.go:124]: Like putting the block
```
                       advNode, err := advertisingNodeFromMAC(epNodes, ingressIP, executor.Host)
			framework.ExpectNoError(err)

			for i := 0; i < 5; i++ {
				name, err := getEndpointHostName(hostport, executor.Host)
				framework.ExpectNoError(err)

				ginkgo.By(fmt.Sprintf("checking that pod %s is on node %s", name, advNode.Name))
				pod, err := cs.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), name, metav1.GetOptions{})
				framework.ExpectNoError(err)
				framework.ExpectEqual(pod.Spec.NodeName, advNode.Name, fmt.Sprintf("traffic arrived to a pod on node %s which is not the announcing node %s", pod.Spec.NodeName, advNode.Name))
			}
```

In an eventually, with an interval of 0.5 seconds and max timeout of 5 seconds.
That would make the test execution slower in fast cases.

**Comment by fedepaol** [comment]
LGTM

**Pull request merged**